src/app/complex/complex-results/complex-navigator/complex-navigator-rating/complex-navigator-rating.component.ts

```diff
@@ -0,0 +1,33 @@
+import {Component} from '@angular/core';
+import {AnalyticsService} from '../../../../shared/google-analytics/service/analytics.service';
+
+@Component({
+  selector: 'cp-complex-navigator-rating',
+  templateUrl: './complex-navigator-rating.component.html',
+  styleUrls: ['./complex-navigator-rating.component.css']
+})
+export class ComplexNavigatorRatingComponent {
+  selectedRating: number;
+  visible = true;
+
+  constructor(private analyticsService: AnalyticsService) {
+  }
+
+  selectRating(rating: number) {
+    this.selectedRating = rating;
+    console.log(`Selected rating: ${rating}`);
+  }
+
+  submitRating() {
+    if (this.selectedRating !== null) {
```

Comment:
Replace with
```
if (!!this.selectedRating) {
```

Initially `selectedRating` is undefined, not null (typescript differentiate between these two values), so because selectedRating is not null, it goes in this if and tries to do selectedRating.toString(), which results in 
```
ERROR TypeError: Cannot read properties of undefined (reading 'toString')
    at ComplexNavigatorRatingComponent.submitRating (complex-navigator-rating.component.ts:23:70)
```

`!!this.selectedRating` is equivalent to write `this.selectedRating != null and this.selectedRating != undefined`.

The only thing that can be tricky with !! is that it is more than just null and undefined, 0 may be also one of the values considered. But, because we don't have a 0 rating value, then it's ok.